File: Project/Android/app/src/main/java/com/example/recetapp/model/recipe/Recipe.kt
package com.example.recetapp.model.recipe

import android.os.Parcelable
import androidx.room.Entity
import androidx.room.PrimaryKey
import com.example.recetapp.model.recipe.ingredients.ExtendedIngredient
import com.example.recetapp.model.recipe.instructions.AnalyzedInstruction
import com.example.recetapp.model.view.CarouselRecipe
import com.google.gson.annotations.SerializedName
import kotlinx.parcelize.Parcelize

@Parcelize
@Entity
data class Recipe(
    @PrimaryKey(autoGenerate = true)
    val id: Int?,
    val title: String?,
    @SerializedName("image")
    val imageUrl: String?,
    @SerializedName("readyInMinutes")
    val timeOfPreparation: Int?,
    val summary: String?,
    val cheap: Boolean?,
    val pricePerServing: Double?,
    val servings: Int?,
    val analyzedInstructions: List<AnalyzedInstruction>?,
    val diets: List<String>?,
    val dishTypes: List<String>?,
    val extendedIngredients: List<ExtendedIngredient>?,
    val instructions: String?,
    var isFavorite: Boolean = false
): Parcelable

fun Recipe.toCarouselRecipe() =
    CarouselRecipe(
        id,
        title,
        imageUrl,
        timeOfPreparation,
        summary,
        cheap,
        pricePerServing,
        servings,
        analyzedInstructions,
        diets,
        dishTypes,
        extendedIngredients,
        instructions
    )

File: Project/iOS/RecetApp/RecetApp/Model/Recipe.swift
//
//  Recipe.swift
//  RecetApp
//
//  Created by Andrea García Ruiz on 22/03/23.
//

import Foundation

// MARK: - Welcome
struct Recipe: Codable {
    let id: Int
    let title: String
    let image: String
    let servings, readyInMinutes: Int
    let pricePerServing: Double
    let analyzedInstructions: [AnalyzedInstruction]
    let diets: [String]
    let instructions: String
    let dishTypes: [String]
    let extendedIngredients: [ExtendedIngredient]
    let summary: String
}

// MARK: - AnalyzedInstruction
struct AnalyzedInstruction: Codable {
    let name: String
    let steps: [Step]
}

// MARK: - Step
struct Step: Codable {
    let number: Int
    let step: String
    let ingredients, equipment: [StepTool]
}

// MARK: - StepEquipment
struct StepTool: Codable {
    let id: Int
    let name, image: String
}

// MARK: - ExtendedIngredient
struct ExtendedIngredient: Codable {
    let id: Int
    let image: String
    let name, original: String
}

enum Consitency: String, Codable {
    case liquid = "liquid"
    case solid = "solid"
}

// MARK: - Measures
struct Measures: Codable {
    let metric, us: Metric
}

// MARK: - Metric
struct Metric: Codable {
    let amount: Double
    let unitLong, unitShort: String
}
